whereby/sdk
# core: Add notification system

### Description

Add an events system to dispatch notification messages to the `state.events` object in the `useRoomConnection` hook that implements the standard [EventEmitter API](https://nodejs.org/dist/v11.13.0/docs/api/events.html#events_class_eventemitter).

### Testing

1. Run `yarn build && yarn dev`
2. Navigate to the "Room Connection Only" story (direct link when storybook is running is [here](http://localhost:6006/?path=/story/examples-custom-ui--room-connection-only)).
3. Observe the "Notifications Log" at the bottom of the UI
4. In the developer tools, toggle the Network connection from "No throttling" to "Offline"
5. After a couple of seconds, toggle the Network connection back from "Offline" to "No throttling"
6. The UI should re-connect to the call and a log of the `connected => disconnected => reconnecting => connected` flow should now be shown in the "Notifications Log" in the UI.

### Screenshots/GIFs (if applicable)

<img width="856" alt="Screenshot 2024-05-03 at 13 16 38" src="https://github.com/whereby/sdk/assets/119658286/c199f3aa-c9a5-401c-9179-6afb27deb241">

### Checklist

        under the terms of the MIT license.
